internal/schema/project.go

```diff
@@ -30,16 +29,16 @@ func (m *ProjectMetadata) WorkspaceLabel() string {
 	return m.TerraformWorkspace
 }
 
-func (m *ProjectMetadata) GenerateProjectName(repoURL string, dashboardEnabled bool) string {
-	// If the VCS repo is set, create the name from that
-	if repoURL != "" {
-		n := nameFromRepoURL(repoURL)
+func (m *ProjectMetadata) GenerateProjectName(remote *vcs.Remote, dashboardEnabled bool) string {
+	// If the VCS repo is set, use its name
+	if remote != nil && remote.Name != "" {
+		name := remote.Name
```

Comment:
why are we using a pointer here? It seems fine just to check that the `remote.Name` is not an empty string?

Reply:
I was cautious in case the remote is not initialized. But I guess if metadata is present, then remote should always be available. I'll change to pass the value instead of the pointer then 👍